### PR TITLE
Feature/add ms teams backend

### PR DESCRIPTION
**Pull request by jsdalton**
## Purpose

The bot needs a backend context to send notifications and handle authentication tasks from

## Approach

* Add a super basic backend to support hosted app actions
* Add a simple test action and script

TODO (future PRs)

* [ ] Add a top level build that packages up the frontend and backend actions in a single artifact to deploy

### CODE REVIEW

**Review by matthew-gordon — APPROVED** [review]
LGTM ✅

File: apps/microsoft-teams/backend/package.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
+{
+  "name": "@contentful/microsoft-teams-app",
```
Comment (whitelisab) [inline apps/microsoft-teams/backend/package.json:2]: small nit: maybe call this `@contentful/microsoft-teams-backend` to mirror what we have in the frontend. Then the  top level `package.json` could be named `@contentful/microsoft-teams-app` or `@contentful/microsoft-teams`

File: apps/microsoft-teams/backend/src/actions/send-test-notification.spec.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+      parameters: {},
+    },
+  ];
+
+  beforeEach(() => {
+    cmaRequestStub = sinon.stub();
+    context = makeMockAppActionCallContext(cmaClientMockResponses, cmaRequestStub);
+  });
+
+  it('returns the images result', async () => {
```
Comment (whitelisab) [inline apps/microsoft-teams/backend/src/actions/send-test-notification.spec.ts:32]: This is the assertion nit that @matthew-gordon mentioned that should be updated eventually.

**Review by whitelisab — APPROVED** [review]
Just left some very small comments that can be addressed later.

**Pull request merged**